### PR TITLE
Suppress warnings with nvc++

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

File: core/src/impl/Kokkos_Error.hpp [hunk 1]
```diff
@@ -79,6 +79,10 @@ namespace Impl {
 #define KOKKOS_IMPL_ABORT_NORETURN [[noreturn]]
 #endif
 
+#elif defined(KOKKOS_COMPILER_NVHPC)
+
+#define KOKKOS_IMPL_ABORT_NORETURN
+
```
Comment (Rombur) [inline core/src/impl/Kokkos_Error.hpp:85]: That should be taken care by the `else`, right.

File: core/src/impl/Kokkos_SharedAlloc.hpp [hunk 2]
```diff
@@ -146,10 +146,17 @@ class SharedAllocationRecord<void, void> {
  public:
   virtual std::string get_label() const { return std::string("Unmanaged"); }
 
+#if defined(__EDG__) && !defined(KOKKOS_COMPILER_INTEL)
```
Comment (Rombur) [inline core/src/impl/Kokkos_SharedAlloc.hpp:149]: Why aren't you using `KOKKOS_COMPILER_NVHPC` instead? There is another compiler with the same problem?

Comment (dalg24) [reply core/src/impl/Kokkos_SharedAlloc.hpp:149]: That's fair.
I was going for consistency with https://github.com/kokkos/kokkos/blob/2ad88ec38922556d4459c4dbd2813845f4b93356/core/src/impl/Kokkos_BitOps.hpp#L98-L101

File: core/src/impl/Kokkos_Error.hpp (hunk 1 above)
Comment (dalg24) [reply core/src/impl/Kokkos_Error.hpp:85]: No that would go through the
```C++
#elif !defined(KOKKOS_ENABLE_OPENMPTARGET)
// Host aborts
#define KOKKOS_IMPL_ABORT_NORETURN [[noreturn]]

**Review by crtrott — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]

**Pull request merged**